cmds/exp/smn/main.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// smn: read or write registers in the System Management Network on AMD cpus
+//
+// Synopsis:
+//     snm [0 or more addresses]
+// N.B. having no addresses is an easy way to see if you can
+// access PCI at all.
+//
+// Description:
+//     read and write System Management Network registers
+//
+// Options:
+//	-s: device glob in the form tttt:bb:dd.fn with * as needed
+//	-n: number of 32-bit words to dump.
+//	-v: 32-bit value to write
+//	-w: write the value to the register(s)
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/pci"
+)
+
+// Index/Data register pairs, as, e.g., cf8/cfc for PCI,
+// are known to be a terrible idea, from almost any point of view.
+// It took years to kill them on regular PCI.
+// AMD brought them back for the SMN. Bummer.
+const (
+	regIndex = 0xa0
+	regData  = 0xa4
+)
+
+var (
+	devs  = flag.String("s", "0000:00:00.0", "Glob for northbridge")
```

Comment:
is all 0s normally the address for the northbridge on amd?

Reply:
on pretty much everything. Thank goodness it's one thing we can count on.